dir.c

```diff
@@ -155,10 +182,17 @@ need_normalization(DIR *dirp, const char *path)
 # if defined HAVE_FGETATTRLIST || defined HAVE_GETATTRLIST
     u_int32_t attrbuf[SIZEUP32(fsobj_tag_t)];
     struct attrlist al = {ATTR_BIT_MAP_COUNT, 0, ATTR_CMN_OBJTAG,};
+    struct getattrlist_args args;
+    args.list = &al;
+    args.buf = attrbuf;
+    args.size = sizeof(attrbuf);
+    args.options = 0;
 #   if defined HAVE_FGETATTRLIST
-    int ret = fgetattrlist(dirfd(dirp), &al, attrbuf, sizeof(attrbuf), 0);
+    args.fd = dirfd(dirp);
+    int ret = IO_WITHOUT_GVL_INT(nogvl_fgetattrlist, &args);
 #   else
-    int ret = getattrlist(path, &al, attrbuf, sizeof(attrbuf), 0);
+    args.path = path;
+    int ret = IO_WITHOUT_GVL_INT(nogvl_getattrlist, &args);
```

Comment:
Rather than repeating similar calls, what about wrapper functions?

Reply:
Pushed a commit to use wrapper functions.  Assuming it passes CI, I'll squash merge it.